### PR TITLE
multiple feeds - Fixed an issue where the *First Seen By source* and …

### PR DESCRIPTION
**Pull request by bakatzir**
…*Last Seen By source* indicator fields were wrongfully populated

## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/34076

## Screenshots
Screen Shot 2021-02-23 at 18 43 14

## Minimum version of Demisto
- [x] 5.5.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests

### CODE REVIEW

**Review by yuvalbenshalom — COMMENTED** [review]
looks good, see my comment

File: Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py [hunk 1]
```diff
@@ -280,9 +280,14 @@ def custom_fields_creator(self, attributes: dict):
         return created_custom_fields
 
 
-def datestring_to_millisecond_timestamp(datestring):
-    date = parse(str(datestring))
-    return int(date.timestamp() * 1000)
+def datestring_formatter(date_string: str) -> str:
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py:283]: I think it is better to format everything to UTC, regardless of the feed, so the IOCs will all show in the indicators tab in the same way.

File: Packs/FeedBlocklist_de/ReleaseNotes/1_0_5.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Blocklist_de Feed
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedBlocklist_de/ReleaseNotes/1_0_5.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/FeedBruteForceBlocker/ReleaseNotes/1_0_5.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### BruteForceBlocker Feed
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedBruteForceBlocker/ReleaseNotes/1_0_5.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/FeedCloudflare/ReleaseNotes/1_0_5.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Cloudflare Feed
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedCloudflare/ReleaseNotes/1_0_5.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/FeedDShield/ReleaseNotes/1_0_5.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### DShield Feed
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedDShield/ReleaseNotes/1_0_5.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/FeedFeodoTracker/ReleaseNotes/1_0_8.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Feodo Tracker Hashes Feed (Deprecated)
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedFeodoTracker/ReleaseNotes/1_0_8.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/FeedMalwareDomainList/ReleaseNotes/1_0_5.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Malware Domain List Active IPs Feed
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedMalwareDomainList/ReleaseNotes/1_0_5.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/FeedPlainText/ReleaseNotes/1_0_6.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Plain Text Feed
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedPlainText/ReleaseNotes/1_0_6.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/FeedSpamhaus/ReleaseNotes/1_0_5.md [hunk 9]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Spamhaus Feed
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedSpamhaus/ReleaseNotes/1_0_5.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/FeedTorExitAddresses/ReleaseNotes/1_0_3.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Tor Exit Addresses Feed
+- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were wrongfully populated.
```
Comment (amshamah419) [inline Packs/FeedTorExitAddresses/ReleaseNotes/1_0_3.md:4]:
```suggestion
- Fixed an issue where the *First Seen By source* and *Last Seen By source* indicator fields were incorrectly populated.
```

File: Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (glicht) [reply Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py:283]: I would rename this to `datestring_to_server_format`. And document that this is the format the server expects to receive.

**Review by glicht — APPROVED** [review]
Please test on a server to make sure that everything works as expected.

**Comment by bakatzir** [comment]
@glicht I tested on latest master. should I spin up 6.0 and 5.5 or is it redundant?

**Review by yuvalbenshalom — APPROVED** [review]

**Comment by bakatzir** [comment]
Tested on 6.0.2 and latest master.
Screen Shot 2021-02-27 at 14 38 17
Screen Shot 2021-02-27 at 14 38 47

**Pull request merged**